include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -52,15 +52,17 @@ namespace seqan3::detail
 {
 
 /*!\brief Implementation of a random access iterator on an input container pointer.
- *  \details No iterator operation will modify the container. Arithmetic and boolean
- *  operations are applied to the iterator positions, not the corresponding values
- *  of their containers.
- *  \tparam container_type The Datastructure on which the iterator operates, e.g. std::vector<int>.
- *  The container_type must satisfy the seqan3::random_access_range_concept and
- *  the seqan3::sized_range_concept.
+ * \tparam container_type The data structure on which the iterator operates, e.g. `std::vector<int>`.
+ *
+ * No iterator operation will modify the container. Arithmetic and boolean
+ * operations are applied to the iterator positions, not the corresponding values
+ * of their containers.
+ *
+ * The iterator makes certain assumptions on the `container_type`, but does not formally require
+ * it to satisfy the seqan3::random_access_range_concept, because the iterator itself is
+ * a requirement for this.
  */
 template <typename container_type>
-    requires random_access_range_concept<container_type> && sized_range_concept<container_type>
```

Comment:
because the random access iterator itself is required for the container to satisfy the container concept → circular dependency